interface.h

```diff
@@ -64,150 +67,853 @@
+ */
 void store_shared_data_to_flash(shared_data_t *shared);
 
+/**
+ * @brief Load a previously stored preset from flash.
+ * 
+ * @param index Index value of the preset to load. A valid value is between 0
+ *        and PRESET_COUNT - 1.
+ * @param preset Pointer to load preset data to.
+ */
 void load_preset_from_flash(uint8_t index, preset_data_t *preset);
+
+/**
+ * @brief Load previously stored preset meta data from flash.
+ * 
+ * @param index Index value of the preset meta data to load. A valid value is
+ *        between 0 and PRESET_COUNT - 1.
+ * @param meta Pointer to load preset meta data to.
+ */
 void load_preset_meta_from_flash(uint8_t index, preset_meta_t *meta);
-void load_shared_data_from_flash(shared_data_t *shared);
 
+/**
+ * @brief Load previously stored shared application data from flash.
+ * 
+ * @param shared Pointer to load shared application data to.
+ */
+void load_shared_data_from_flash(shared_data_t *shared);
 
 // ----------------------------------------------------------------------------
 // screen
 
+/**
+ * @brief Clears all lines of the screen. Call `refresh_screen()` to display the
+ *        results. Applies to Teletype module.
+ * 
+ * @see refresh_screen()
+ */
 void clear_screen(void);
+
+/**
+ * @brief Fills a line of the screen with the indicated color. Call
+ *        `refresh_screen()` to display the results. Applies to Teletype module.
+ * 
+ * @param line Line number to fill. A valid value is between 0 and
+ *        SCREEN_LINE_COUNT - 1.
+ * @param colour Colour value to fill. A valid value is between 0 and 15,
+ *        0 = off, 15 = full level
+ * 
+ * @see refresh_screen()
+ */
 void fill_line(uint8_t line, uint8_t colour);
+
+/**
+ * @brief Draws a string value to the screen at the indicated line. Call
+ *        `refresh_screen()` to display the results. Applies to Teletype module.
+ * 
+ * @param str String value to draw
+ * @param line Line number of screen to draw to. A valid value is between 0 and
+ *        SCREEN_LINE_COUNT - 1
+ * @param colour Colour value to draw the string. A valid value is between 0 and
+ *        15, 0 = off, 15 = full level
+ * @param background Background colour value to draw the string on top of. A
+ *        valid value is between 0 and 15, 0 = off, 15 = full level
+ * 
+ * @see refresh_screen()
+ */
 void draw_str(const char* str, uint8_t line, uint8_t colour, uint8_t background);
+
+/**
+ * @brief Refresh the displayed content of the screen. Applies to Teletype
+ *        module.
+ * 
+ */
 void refresh_screen(void);
 
 
 // ----------------------------------------------------------------------------
 // other
 
+/**
+ * @brief Set the state of the LEDs on the Monome hardware. Ansible is the only
+ *        module that has individually addressable LEDs.
+ * 
+ * @param index Hardware LED number. A valid value is between 0 and
+ *        _HARDWARE_LED_COUNT - 1.
+ * @param level FIXME: Unclear how to describe this. 
```

Comment:
i'm going to change it to `colour` i think - this is one of those cases where abstracting it doesn't make much sense, but at least it'll be consistent with the screen function..

Reply:
i believe that's correct. i tried to make non module specific functions more generic, so this why it's `level` parameter instead of something ansible specific - this is the kind of info that would be good to put in a separate "module specifics" topic.

Reply:
✔️